indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -621,4 +653,96 @@ private static DimensionSchema createDimensionSchema(
       return usedSegments;
     }
   }
+
+  @VisibleForTesting
+  static class PartitionConfigurationManager
+  {
+    @Nullable
+    private final Long targetCompactionSizeBytes;
+    @Nullable
+    private final IndexTuningConfig tuningConfig;
+
+    PartitionConfigurationManager(@Nullable Long targetCompactionSizeBytes, @Nullable IndexTuningConfig tuningConfig)
+    {
+      this.targetCompactionSizeBytes = getValidTargetCompactionSizeBytes(targetCompactionSizeBytes, tuningConfig);
+      this.tuningConfig = tuningConfig;
+    }
+
+    @Nullable
+    IndexTuningConfig computeTuningConfig(List<Pair<QueryableIndex, DataSegment>> queryableIndexAndSegments)
+    {
+      if (!hasPartitionConfig(tuningConfig)) {
+        // Find IndexTuningConfig.targetPartitionSize which is the number of rows per segment.
+        // Assume that the segment size is proportional to the number of rows. We can improve this later.
+        final double avgRowsPerByte = queryableIndexAndSegments
+            .stream()
+            .mapToDouble(queryableIndexAndDataSegment -> {
+              final long numRows = queryableIndexAndDataSegment.lhs.getNumRows();
```

Comment:
this is taking an average of averages, would it be better to do one average for (total num rows / total size)

Reply:
Fixed.